From 800eb172195a30b898d00ad483d941ab339ffdc2 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 25 Sep 2014 11:05:01 +0800
Subject: [PATCH 0310/1594] MLK-9634-1 arm: imx: remove romcp workaround for
 i.mx6sx TO1.2

commit a71f68cfbb979b3b2aee4dc5aecb675480070720 from
git://git.freescale.com/imx/linux-2.6-imx.git

i.MX6SX TO1.2 ROM has add ocram_s space support for ARM resume,
so no need to enable ROMCP to workaround it for TO1.2.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/pm-imx6.c |   32 ++++++++++++++++++--------------
 1 files changed, 18 insertions(+), 14 deletions(-)

diff --git a/arch/arm/mach-imx/pm-imx6.c b/arch/arm/mach-imx/pm-imx6.c
index a1a05f3..fb46a50 100644
--- a/arch/arm/mach-imx/pm-imx6.c
+++ b/arch/arm/mach-imx/pm-imx6.c
@@ -747,26 +747,30 @@ void __init imx6sl_pm_init(void)
 void __init imx6sx_pm_init(void)
 {
 	imx6_pm_common_init(&imx6sx_pm_data);
+	if (imx_get_soc_revision() < IMX_CHIP_REVISION_1_2) {
 	/*
 	 * As there is a 16K OCRAM(start from 0x8f8000)
 	 * dedicated for low power function on i.MX6SX,
 	 * but ROM did NOT do the ocram address change
 	 * accordingly, so we need to add a data patch
 	 * to workaround this issue, otherwise, system
-	 * will fail to resume from DSM mode.
+	 * will fail to resume from DSM mode. TO1.2 fixes
+	 * this issue.
 	 */
-	romcp = syscon_regmap_lookup_by_compatible("fsl,imx6sx-romcp");
-	if (IS_ERR(romcp)) {
-		pr_err("failed to find fsl,imx6sx-romcp regmap\n");
-		return;
+		romcp = syscon_regmap_lookup_by_compatible(
+			"fsl,imx6sx-romcp");
+		if (IS_ERR(romcp)) {
+			pr_err("failed to find fsl,imx6sx-romcp regmap\n");
+			return;
+		}
+		regmap_write(romcp, ROMC_ROMPATCH0D, iram_tlb_phys_addr);
+		regmap_update_bits(romcp, ROMC_ROMPATCHCNTL,
+			BM_ROMPATCHCNTL_0D, BM_ROMPATCHCNTL_0D);
+		regmap_update_bits(romcp, ROMC_ROMPATCHENL,
+			BM_ROMPATCHENL_0D, BM_ROMPATCHENL_0D);
+		regmap_write(romcp, ROMC_ROMPATCH0A,
+			ROM_ADDR_FOR_INTERNAL_RAM_BASE);
+		regmap_update_bits(romcp, ROMC_ROMPATCHCNTL,
+			BM_ROMPATCHCNTL_DIS, ~BM_ROMPATCHCNTL_DIS);
 	}
-	regmap_write(romcp, ROMC_ROMPATCH0D, iram_tlb_phys_addr);
-	regmap_update_bits(romcp, ROMC_ROMPATCHCNTL,
-		BM_ROMPATCHCNTL_0D, BM_ROMPATCHCNTL_0D);
-	regmap_update_bits(romcp, ROMC_ROMPATCHENL,
-		BM_ROMPATCHENL_0D, BM_ROMPATCHENL_0D);
-	regmap_write(romcp, ROMC_ROMPATCH0A,
-		ROM_ADDR_FOR_INTERNAL_RAM_BASE);
-	regmap_update_bits(romcp, ROMC_ROMPATCHCNTL,
-		BM_ROMPATCHCNTL_DIS, ~BM_ROMPATCHCNTL_DIS);
 }
-- 
1.7.5.4

